src/compute/flexbox.rs

```diff
@@ -1733,24 +1737,43 @@ fn calculate_flex_item(
 
     if direction.is_row() {
         let baseline_offset_cross = total_offset_cross + item.offset_cross + item.margin.cross_start(direction);
-        let inner_baseline = preliminary_size_and_baselines.first_baselines.y.unwrap_or(preliminary_size.height);
+        let inner_baseline = layout_output.first_baselines.y.unwrap_or(size.height);
         item.baseline = baseline_offset_cross + inner_baseline;
     } else {
         let baseline_offset_main = *total_offset_main + item.offset_main + item.margin.main_start(direction);
-        let inner_baseline = preliminary_size_and_baselines.first_baselines.y.unwrap_or(preliminary_size.height);
+        let inner_baseline = layout_output.first_baselines.y.unwrap_or(size.height);
         item.baseline = baseline_offset_main + inner_baseline;
     }
 
-    *tree.get_unrounded_layout_mut(item.node) = Layout {
-        order: item.order,
-        size: preliminary_size_and_baselines.size,
-        location: Point {
-            x: if direction.is_row() { offset_main } else { offset_cross },
-            y: if direction.is_column() { offset_main } else { offset_cross },
-        },
+    let location = Point {
+        x: if direction.is_row() { offset_main } else { offset_cross },
+        y: if direction.is_column() { offset_main } else { offset_cross },
     };
```

Comment:
Perhaps easier and only needs one check (will hardly make a perf difference though):
```suggestion
    let location = if direction.is_row() {
        Point {
            x: offset_main,
            y: offset_cross,
        }
    } else {
        Point {
            x: offset_cross,
            y: offset_main,
        }
    };
```

Reply:
I've gone with
```rust
let location = match constants.is_row {
      true => Point { x: offset_main, y: offset_cross },
      false => Point { x: offset_cross, y: offset_main },
};
```
to keep it short :)